src/adapt.ts

```diff
@@ -349,11 +399,15 @@ export function adaptOutputUtil(
 
     for (const outputSpec of outputSpecs) {
         const currentValue = result[resultIndex];
-        if (COMMON_TYPES.includes(outputSpec.type)) {
+        if (COMMON_NUMERIC_TYPES.includes(outputSpec.type)) {
             outputSpec.name = outputSpec.name ?? "response";
             adapted[outputSpec.name] = currentValue;
             resultIndex++;
-        } else if (outputSpec.type.endsWith("*")) {
+        } else if (isBool(outputSpec.type)) {
+            outputSpec.name = outputSpec.name ?? "response";
```

Comment:
In some places we modify the `outputSpec` object, in some we don't. Could this logic be abstracted to a common function? In any case, this ought to be better addressed in the future, so I've created this issue: https://github.com/0xSpaceShard/starknet-hardhat-plugin/issues/356